### PR TITLE
Simplify the EventDispatcher

### PR DESCRIPTION
**Pull request by BackEndTea**
This PR:

- [x] Removes dead code from the EventDispatcher

Most of its public functions were not being used within the code base(anymore).
Removed the unused functions, the now useless interface,and updated tests to show this behaviour.

### CODE REVIEW

File: src/EventDispatcher/EventDispatcher.php [hunk 1]
```diff
@@ -31,41 +31,24 @@ public function dispatch($event)
...
-     * {@inheritdoc}
+     * @param string $eventName
+     *
+     * @return callable[]
      */
-    public function getListeners($eventName)
+    private function getListeners($eventName)
     {
-        if (!$this->hasListeners($eventName)) {
+        if (!isset($this->listeners[$eventName])) {
```
Comment (sidz) [inline src/EventDispatcher/EventDispatcher.php]: can be simplified as well: 
`return $this->listeners[$eventName] ?? [];`

File: src/EventDispatcher/EventDispatcher.php [hunk 2]
```diff
@@ -31,41 +31,24 @@ public function dispatch($event)
...
     }
 
     /**
-     * {@inheritdoc}
+     * @param string $eventName
+     *
+     * @return callable[]
      */
-    public function getListeners($eventName)
+    private function getListeners($eventName)
```
Comment (sidz) [inline src/EventDispatcher/EventDispatcher.php]: seems like we can add `string` type hinting here

**Comment by borNfreee** [comment]
Thank you @BackEndTea

**Comment by BackEndTea** [comment]
Thank you @borNfreee @sidz

**Pull request merged**